### ISSUE TITLE
Asks for CMakeLists.txt but build failed with -2

### ISSUE THREAD

**Issue by vinodappu** [microsoft/vscode-cmake-tools#3554]
### Brief Issue Summary

In a workspace similar to what is described in the issue [Multiple CMakeLists.txt without Requiring a Multi-Root Workspace](https://github.com/microsoft/vscode-cmake-tools/issues/1374#top) & no cmake.sourceDirectory is not configured, when user tries to camke.configure, cmake-tools asks for CmakeLists.txt but when tries to build, nothing happening. If I close and reopen the same workspace, the build is happening!
![cmake-tools-build-with-cmakelists-selection-failed](https://github.com/microsoft/vscode-cmake-tools/assets/65697683/28e3ae4c-09aa-4369-8eac-88fd5e1dde3a)
![cmake-tools-build-with-cmakelists-selection-failed-log](https://github.com/microsoft/vscode-cmake-tools/assets/65697683/dd9cc544-91e0-46ba-9425-8b0bc0435533)

### CMake Tools Diagnostics

```shell
{
  "os": "linux",
  "vscodeVersion": "1.85.2",
  "cmtVersion": "2.13.0",
  "configurations": [
    {
      "folder": "/home/vinod/vsc_wsl_root_test",
      "cmakeVersion": "3.27.1",
      "configured": false,
      "generator": "Unix Makefiles",
      "usesPresets": false,
      "compilers": {}
    }
  ],
  "cpptoolsIntegration": {
    "isReady": false,
    "hasCodeModel": false,
    "activeBuildType": "",
    "buildTypesSeen": [],
    "requests": [],
    "responses": [],
    "partialMatches": [],
    "targetCount": 0,
    "executablesCount": 0,
    "librariesCount": 0,
    "targets": []
  },
  "settings": [
    {
      "communicationMode": "automatic",
      "useCMakePresets": "auto",
      "configureOnOpen": false
    }
  ]
}
```

### Debug Log

```shell
[main] Building folder: vsc_wsl_root_test 
[expand] expanding /home/vinod/.ashling/23.4.2/build_tools/cmake/bin/cmake
[main] Saving open files before configure/build
[expand] expanding /home/vinod/.ashling/23.4.2/build_tools/cmake/bin/cmake
[expand] expanding /home/vinod/.ashling/23.4.2/build_tools/cmake/bin/cmake
[main] Configuring project: vsc_wsl_root_test 
[main] Saving open files before configure/build
[expand] expanding /home/vinod/vsc_wsl_root_test
[expand] expanding ${sourceDirectory}/build
[driver] Start configure 
[driver] Running pre-configure checks and steps
[driver] Not configuring: There is no /home/vinod/vsc_wsl_root_test/CMakeLists.txt
[expand] expanding ${workspaceFolder}/helloworld-cmake
[expand] expanding /home/vinod/.ashling/23.4.2/build_tools/cmake/bin/cmake
[extension] [5737] cmake.build finished (returned -2)
```

**Comment by gcampbell-msft** [comment]
@vinodappu Thanks for posting this! This is definitely a bug we'd want to fix, I've placed it into the `On Deck` category, which is where we pull bugs from when picking bugs to solve for the next release.

Thanks!

**Comment by vinodappu** [comment]
![image](https://github.com/microsoft/vscode-cmake-tools/assets/65697683/85945ce8-14d1-4da7-9ccb-d11308f4a6cc)

In cmakeDriver.ts, this line seems the issue. We resolve everything but returning false.

**Comment by gcampbell-msft** [comment]
@vinodappu I'm trying to understand exactly what step you're saying this issue crops up in. Could you provide explicit repro steps?

**Comment by gcampbell-msft** [comment]
@vinodappu If you wanted to test, we have a fix in PR here: #3776

**Comment by v-frankwang** [comment]
@vinodappu We tried to verify this issue on CMake tools: v1.19.45, but we don't have detailed reproduction steps to reproduce it. Are you able to go and verify that the issue is fixed?
